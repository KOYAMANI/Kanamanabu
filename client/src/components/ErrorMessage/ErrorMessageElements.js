
import styled from 'styled-components';
import { Alert } from 'react-bootstrap'

export const ErrorAlert = styled(Alert)`
    fontSize: 30;
    variant: ${({ message }) => message || "Error"};,
`
