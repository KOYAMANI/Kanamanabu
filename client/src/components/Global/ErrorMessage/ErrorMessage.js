
import React from 'react'
import { ErrorAlert } from './ErrorMessageElements'

const ErrorMessage = ({
    variant= "info", 
    children
}) => {
    return (
        <ErrorAlert 
        variant = {variant}>
            <strong>{children}</strong>
        </ErrorAlert>
    )
}

export default ErrorMessage