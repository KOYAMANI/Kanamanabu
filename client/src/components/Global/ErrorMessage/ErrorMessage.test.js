import {render, screen, cleanup} from '@testing-library/react';
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer'
import ErrorMessage from "./ErrorMessage";

afterEach(cleanup)

describe('ErrorMessage', () =>{
    test('Render without crashing', ()=>{
        const div = document.createElement('div')
        ReactDOM.render(<ErrorMessage>test</ErrorMessage>, div)
    });

    test('Render text', ()=>{
        render(<ErrorMessage variant="info">error-message-test</ErrorMessage>)
        const element = screen.getByText('error-message-test')
        expect(element).toBeInTheDocument();
        expect(element).toHaveTextContent('error-message-test')
    })

    test('match snapshot',()=>{
        const tree = renderer.create(<ErrorMessage>test</ErrorMessage>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})