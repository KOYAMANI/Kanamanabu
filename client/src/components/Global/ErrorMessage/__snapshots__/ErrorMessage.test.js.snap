// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage match snapshot 1`] = `
<div
  className="fade sc-bczRLJ knBbvx alert alert-info show"
  role="alert"
>
  <strong>
    test
  </strong>
</div>
`;
